src/OverlayTrigger.js

```diff
@@ -77,192 +96,164 @@ const propTypes = {
 };
 
 const defaultProps = {
-  defaultOverlayShown: false,
+  defaultShow: false,
   trigger: ['hover', 'focus'],
 };
 
-// eslint-disable-next-line react/no-multi-comp
-class OverlayTrigger extends React.Component {
-  constructor(props, context) {
-    super(props, context);
-
-    this.trigger = React.createRef();
-    this.state = {
-      show: !!props.defaultShow,
-    };
-
-    // We add aria-describedby in the case where the overlay is a role="tooltip"
-    // for other cases describedby isn't appropriate (e.g. a popover with inputs) so we don't add it.
-    this.ariaModifier = {
-      enabled: true,
-      order: 900,
-      fn: data => {
-        const { popper } = data.instance;
-        const target = this.getTarget();
-        if (!this.state.show || !target) return data;
-
-        const role = popper.getAttribute('role') || '';
-        if (popper.id && role.toLowerCase() === 'tooltip') {
-          target.setAttribute('aria-describedby', popper.id);
-        }
-        return data;
-      },
-    };
-  }
+function OverlayTrigger({
+  trigger,
+  overlay,
+  children,
+  popperConfig = {},
+  defaultShow,
+  delay: propsDelay,
+  ...props
+}) {
+  const triggerNodeRef = useRef(null);
+  const timeout = useTimeout();
+  const hoverStateRef = useRef();
+  const [show, setShow] = useState(!!defaultShow);
 
-  componentWillUnmount() {
-    clearTimeout(this._timeout);
-  }
+  useWillUnmount(() => timeout.clear());
```

Comment:
oh, we can actually remove this entirely, if that's all we need from unmounting

`useTimeout` will clear itself on unmount

Reply:
It even calls `useWillUnmount` internally.
https://github.com/react-restart/hooks/blob/master/src/useTimeout.ts#L41

Reply:
removed it, thanks!